spec/requests/api/v2/providers/suggest_provider_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+describe 'GET /suggest' do
+  let(:jsonapi_renderer) { JSONAPI::Serializable::Renderer.new }
+  let(:organisation) { create(:organisation) }
+  let(:provider1) { create(:provider, organisations: [organisation]) }
+  let(:provider2) { create(:provider, organisations: [organisation]) }
+  let(:next_recruitment_cycle) { find_or_create(:recruitment_cycle, :next) }
+  let(:provider3) { create(:provider, provider_name: provider2.provider_name, organisations: [organisation], recruitment_cycle: next_recruitment_cycle) }
+  let(:unauthorized_provider) { create(:provider, organisations: [create(:organisation)]) }
+  let(:user) { create :user, organisations: [organisation] }
+  let(:payload) { { email: user.email } }
+  let(:token) { build_jwt :apiv2, payload: payload }
+  let(:credentials) do
+    ActionController::HttpAuthentication::Token.encode_credentials(token)
+  end
+
+  before do
+    provider1
+    provider2
+    provider3
+    unauthorized_provider
+  end
+
+  it 'searches for a particular provider for the user for the current recruitment cycle' do
+    get "/api/v2/providers/suggest?query=#{provider2.provider_name}",
```

Comment:
/api/v2/providers/suggest?query=#{provider2.provider_name[0, n]}

where n is less then 3 fail
where n  + 3 to max succed

Reply:
See https://github.com/DFE-Digital/manage-courses-backend/pull/772/files#diff-117aabed956114d91bb7d529138c0b76R47

Reply:
That's not the same.

I prefer the test to use values from `provider2.provider_name` if its less then 3 chars it should fail 
`Ge` simply isn't the same as its not the subset of `provider2.provider_name` as we known that `provider2.provider_name` is in there where as `Ge` is questionable

if its 3 to the whole name it should return `provider2`

I would expect given a `provider_name` as `Paul's Uni`

fail
``
`P` 
`Pa`

it failes as the query is less then 3 chars from provider2 provider name

passes
`Pau`
`Paul`
`Paul's` 
`Paul's U` 
`Paul's Un`

This was what was tested
`Paul's Uni`

Reply:
It won't make a difference whether the provider exists or not, a query should be rejected before it gets to that stage or you might be querying unnecessarily.

Reply:
It is a bit pointless, but my justification, is that it is more aligned to real usage

A user will start typing and it will start querying the backend.
Until the user stops typing, each char inputted will cause a http request.

A simple test of 
`get "/api/v2/providers/suggest?query=#{provider2.provider_name}"`
only ever occurs with copy and pasting

So the caller may not have validated input
hence that is why I wanted testing using from 0 to the full provider name.

None of the tests here shows the expectation of a query that has the partial of the full provider name, which is just as important as testing for the full provider name.

endpoints that is expected to be called multiple times in a short space of time warrants a bit more test coverage.

I settle for one partial instead of the whole enumeration to getting to the full provider name

Reply:
> It won't make a difference whether the provider exists or not

Not sure what you meant?

```
it 'searches for a non existing provider for the user for the current recruitment cycle' do
get "/api/v2/providers/suggest?query=#{non existing}",
```

if you meant that, it  matters as would of hit the database and 0 rows would of returned

Reply:
Because the database should never be hit if the query is too short.